projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -180,55 +193,62 @@ int32_t spi_init(struct spi_desc **desc,
 		 const struct spi_init_param *param)
 {
 	spi_desc *descriptor;
+	extra_spi_desc *extra_descriptor;
+	extra_spi_init_param *extra_param;
 	int32_t ret;
 
 	descriptor = (struct spi_desc *) calloc(1, sizeof(*descriptor));
-	if (!descriptor)
+	extra_descriptor = (struct extra_spi_desc *) calloc(1,
+			   sizeof(*extra_descriptor));
+	if (!descriptor || !extra_descriptor)
 		return FAILURE;
 
+	descriptor->extra = extra_descriptor;
+	extra_param = param->extra;
```

Comment:
why use an extra variable for param->extra ?

Reply:
using param->extra requires casting each time it is accessed.
In my opinion, adding an intermediary variable will improve the code readability. 

I.e. 
`xil_dev->id = ((xil_i2c_desc *)param->extra)->id;`
vs
`xil_dev->id = xil_param->id;`

Reply:
In this manner we avoid casting as much as possible.